pool_boy.py

```diff
@@ -45,37 +54,103 @@ def refresh_local_repo(conf):
     return git
 
 
+class BearerAuth(requests.auth.AuthBase):
+    def __init__(self, token):
+        self.token = token
+
+    def __call__(self, request):
+        request.headers['Authorization'] = 'Bearer ' + self.token
+        return request
+
+
+def get_concourse_auth():
```

Comment:
Maybe add a link to https://www.oauth.com/oauth2-servers/access-tokens/password-grant/